x-pack/legacy/plugins/reporting/server/types.ts

```diff
@@ -188,15 +191,14 @@ export interface LegacySetup {
  */
 
 export interface RequestFacade {
-  getBasePath: Legacy.Request['getBasePath'];
```

Comment:
Since this interface no longer depends on legacy references, it would be handy to now have it defined in our new platform plugin area, in `x-pack/plugins/reporting/server/types.ts` 

That file doesn't exist yet, but having it created in this PR could help us work in parallel without one of us hitting extra nasty conflicts.